File: problems/problems_3159/Solution.java
package problems.problems_3159;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int[] occurrencesOfElement(int[] nums, int[] queries, int x) {
        
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] nums = jsonArrayToIntArray(inputJsonValues[0]);
		int[] queries = jsonArrayToIntArray(inputJsonValues[1]);
		int x = Integer.parseInt(inputJsonValues[2]);
        return JSON.toJSON(occurrencesOfElement(nums, queries, x));
    }
}

File: problems/problems_3159/solution.go
package problem3159

import (
	"encoding/json"
	"log"
	"strings"
)

func occurrencesOfElement(nums []int, queries []int, x int) []int {
	var idxes []int
	for i, num := range nums {
		if num == x {
			idxes = append(idxes, i)
		}
	}
	ans := make([]int, len(queries))
	for i, query := range queries {
		if query > len(idxes) {
			ans[i] = -1
			continue
		}
		ans[i] = idxes[query-1]
	}
	return ans
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var nums []int
	var queries []int
	var x int

	if err := json.Unmarshal([]byte(inputValues[0]), &nums); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(inputValues[1]), &queries); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(inputValues[2]), &x); err != nil {
		log.Fatal(err)
	}

	return occurrencesOfElement(nums, queries, x)
}
